core/src/ContactEKF.hpp

```diff
@@ -10,101 +10,159 @@
  * You should have received a copy of the GNU General Public License along with Serow. If not,
  * see <https://www.gnu.org/licenses/>.
  **/
+
 /**
- * @brief Base Estimator fusing Inertial Measurement Unit (IMU), relative to the base leg contact
- * measurements and an external odometry e.g Visual Odometry (VO) or Lidar Odometry (LO)
+ * @file ContactEKF.hpp
+ * @brief Implementation of Extended Kalman Filter (EKF) for state estimation in humanoid robots.
+ *        This EKF fuses data from an Inertial Measurement Unit (IMU), relative to the base leg
+ *        contact measurements, and optionally external odometry (e.g., Visual Odometry or Lidar
+ *        Odometry). It models the state of the robot including base position, velocity,
+ * orientation, gyro and accelerometer biases, leg contact positions, and contact orientations in
+ * world frame coordinates.
  * @author Stylianos Piperakis
- * @details State is:
- * Base position in world frame coordinates
- * Base linear velocity in base frame coordinates
- * Base orientation w.r.t the world frame coordinates
- * Gyro bias in base frame coordinates
- * Accelerometer bias in base frame coordinates
- * Leg contact position in world frame coordinates
- * Leg contact orientation in world frame coordinates
- * More info in Nonlinear State Estimation for Humanoid Robot Walking
- * https://www.researchgate.net/publication/326194869_Nonlinear_State_Estimation_for_Humanoid_Robot_Walking
+ * @details The state estimation involves predicting and updating the robot's state based on sensor
+ *          measurements and the robot's dynamics. The EKF integrates outlier detection mechanisms
+ *          to improve state estimation robustness, incorporating beta distribution parameters and
+ *          digamma function approximations. More information on the nonlinear state estimation for
+ *          humanoid robot walking can be found in:
+ *          https://www.researchgate.net/publication/326194869_Nonlinear_State_Estimation_for_Humanoid_Robot_Walking
  */
+
 #pragma once
 
 #include <iostream>
 
-#include "Measurement.hpp"
-#include "State.hpp"
+#include "Measurement.hpp"  // Includes various sensor measurements
+#include "State.hpp"        // Includes definitions of robot state variables
 
 namespace serow {
 
+/**
+ * @struct OutlierDetector
+ * @brief Implements outlier detection mechanisms using beta distribution parameters for state
+ *        estimation in humanoid robots.
+ */
 struct OutlierDetector {
-    /// Beta distribution parameters more info in Outlier-Robust State Estimation for Humanoid Robots
-    /// https://www.researchgate.net/publication/334745931_Outlier-Robust_State_Estimation_for_Humanoid_Robots
-    double zeta = 1.0;
-    double f_0 = 0.1;
-    double e_0 = 0.9;
-    double f_t = 0.1;
-    double e_t = 0.9;
-    double threshold = 1e-5;
-    size_t iters = 4;
-
-    /// Digamma function approximation
+    double zeta = 1.0;        ///< Parameter for outlier detection
```

Comment:
Too many  redundant comments. <Parameter for outlier detection. Either specify a formula or a tip on how to choose these parameters appropriately